airflow/www/app.py

```diff
@@ -272,10 +282,6 @@ def apply_caching(response):
                 response.headers["X-Frame-Options"] = "DENY"
             return response
 
-        @app.teardown_appcontext
```

Comment:
This is now handled correctly by the flask-sqlalchemy library.
https://github.com/pallets/flask-sqlalchemy/blob/master/flask_sqlalchemy/__init__.py#L840-L847